rsmtool/test_utils.py

```diff
@@ -700,3 +706,242 @@ def check_subgroup_outputs(output_dir, experiment_id, subgroups, file_format='cs
             length = len(composition_by_group.loc[composition_by_group['{} set'
                                                                        ''.format(partition)] != 0])
             ok_(length == partition_info.iloc[0][group])
+
+
+class FileUpdater(object):
+
+    """
+    A FileUpdater object is used to update the test outputs
+    for the tests in the `tests_directory` based on the
+    outputs contained in the `updated_outputs_directory`.
+    It does this for all of the experiment tests contained
+    in the test files given by each of the `test_suffixes`.
+
+    Attributes:
+        tests_directory (str): Path to the directory containing the tests whose
+            outputs are to be updated.
+        test_suffixes (list): List of suffixes that will be added to the string
+            "test_experiment_" and located in the `tests_directory` to find
+            the tests that are to be updated.
+        updated_outputs_directory (TYPE): Description
+        deleted_files (list): List of files deleted from `tests directory`.
+        updated_files (list): List of files added or updated for the tests
+            under `tests_directory`.
+        missing_or_empty_sources (list): List of source names whose corresponding
+            directories are either missing under `updated_outputs_directory` or
+            do exist but are empty.
+    """
+
+    def __init__(self,
+                 test_suffixes,
+                 tests_directory,
+                 updated_outputs_directory):
+        self.test_suffixes = test_suffixes
+        self.tests_directory = Path(tests_directory)
+        self.updated_outputs_directory = Path(updated_outputs_directory)
+        self.missing_or_empty_sources = []
+        self.deleted_files = []
+        self.updated_files = []
+
+        # invalidate the file comparison cache
+        clear_cache()
+
+    def is_skll_excluded_file(self, filename):
+        """
+        Checks whether the given filename is one that should
+        be excluded for SKLL-based experiment tests.
+
+        Args:
+            filename (str): Name of the file to be checked.
+
+        Returns:
+            exclude: `True` if the file should be excluded.
+                `False` otherwise.
+        """
+        stem = Path(filename).stem
+        suffix = Path(filename).suffix
+        return suffix == '.model' or \
+            suffix == '.npy'or \
+            stem.endswith('_postprocessing_params') or \
+            stem.endswith('_eval') or \
+            stem.endswith('_eval_short') or \
+            stem.endswith('_confMatrix') or \
+            stem.endswith('_pred_train') or \
+            stem.endswith('_pred_processed') or \
+            stem.endswith('_score_dist')
```

Comment:
This is probably not a big deal, but would we ever be adding to the list of possible `skll` suffixes or stems here? Is it easier to put these into lists, rather than having so many `or` statements?

```
possible_suffixes = ['.model', '.npy']
possible_stems = ['_postprocessing_params', '_eval', '_eval_short', ...]

return any(suffix == s for s in possible_suffixes) or \
    any(stem.endswith(s) for s in possible_stems)
```